source/develop/release-management/features/storage/vm-leases.html.md

```diff
@@ -0,0 +1,458 @@
+---
+title: VM Leases
+category: feature
+authors: nsoffer
+feature_name: VM Leases
+feature_modules: engine,vdsm
+feature_status: Design
+---
+
+# VM Leases
+
+
+## Overview
+
+This feature adds the ability to acquire a lease per VM on shared
+storage without attaching the lease to a disk. With a VM lease, we gain
+two important capabilities; avoiding split-brain, and starting a VM on
+another host if the original host becomes non-responsive.  The later
+capability will be used to improve availability of HA VMs.
+
+
+## Owner
+
+- Name: [Nir Soffer](https://github.com/nirs)
+- Email: <nsoffer@redhat.com>
+
+
+## Detailed Description
+
+
+### How it works
+
+When creating or upgrading a storage domain to version 4, vdsm will
+create a new special volume for external leases, named "xleases". This
+volume will be used for external leases such as VM leases. Vdsm uses
+this volume to create sanlock resources, and maintain the mapping from
+lease id to lease offset on the volume.
```

Comment:
We can, but this introduce several problems
- Now we have new failures modes when we create a lease (cannot create xleases lv)
- We need to activate this volume when a storage domain is activated, and deactivate when storage domain is deactivated. This is already implemented for other special lvs.

We can create a minimal volume of 128MiB, and then extend it as needed, but I don't think it worth the effort for the first version.

I'l describe the creation process and extend policy in the feature page.